aggregation/simple/src/main/java/com/spotify/heroic/aggregation/simple/DeltaInstance.java

```diff
@@ -0,0 +1,142 @@
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package com.spotify.heroic.aggregation.simple;
+import com.spotify.heroic.aggregation.AggregationInstance;
+import com.spotify.heroic.aggregation.AggregationResult;
+import com.spotify.heroic.aggregation.AggregationSession;
+import com.spotify.heroic.aggregation.AggregationOutput;
+import com.spotify.heroic.aggregation.EmptyInstance;
+import com.spotify.heroic.common.DateRange;
+import com.spotify.heroic.common.Series;
+import com.spotify.heroic.metric.*;
+import com.spotify.heroic.metric.Spread;
+import lombok.Data;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Set;
+import java.util.stream.Collectors;
+
+@Data
+public class DeltaInstance implements AggregationInstance {
+    private final EmptyInstance inner = new EmptyInstance();
+
+    @Override
+    public long estimate(DateRange range) {
+        return -1;
+    }
+
+    @Override
+    public long cadence() {
+        return -1;
+    }
+
+    @Override
+    public AggregationInstance distributed() {
+        return this;
+    }
+
+    @Override
+    public boolean distributable() {
+        return false;
+    }
+
+    public List<Point> computeDiff(List<Point> points) {
+        final List<Point> result = new ArrayList<Point>();
+        double lastValue = points.get(0).getValue();
+        for (int i = 0; i < points.size(); i++) {
+            Point thisPoint = points.get(i);
+            double diff = thisPoint.getValue() - lastValue;
+            lastValue = thisPoint.getValue();
+            Point point = new Point(thisPoint.getTimestamp(), diff);
+            result.add(point);
+        }
+        return result;
+    }
+
+    @Override
+    public AggregationSession session(DateRange range) {
+        return new Session(inner.session(range));
+    }
+
+    private class Session implements AggregationSession {
+        private final AggregationSession childSession;
+
+        public Session(AggregationSession childSession) {
+            this.childSession = childSession;
+        }
+
+        @Override
+        public void updatePoints(
+            final Map<String, String> key, final Set<Series> series, final List<Point> values
+        ) {
+            this.childSession.updatePoints(key, series, values);
+        }
+
+        @Override
+        public void updateEvents(
+            final Map<String, String> key, final Set<Series> series, final List<Event> values
+        ) {
+            this.childSession.updateEvents(key, series, values);
+        }
+
+        @Override
+        public void updatePayload(
+            final Map<String, String> key, final Set<Series> series, final List<Payload> values
+        ) {
+            this.childSession.updatePayload(key, series, values);
```

Comment:
Leave this and others empty if you don't intend to support any other metric types. This will cause it to ignore data which is of the wrong type.